File: 2021/01/part1.py
def part1(filename):
	with open(filename, 'r') as file:
		data = [int(l) for l in file.read().splitlines()]
	return sum([1 if data[i] > data[i-1] else 0 for i in range(1, len(data))])

if __name__ == '__main__':
	count = part1('input.txt')
	print(count)

File: 2022/03/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func part1() {
	file, _ := os.Open("input.txt")
	scanner := bufio.NewScanner(file)
	defer file.Close()

	sum := 0
	for scanner.Scan() {
		line := scanner.Text()
		mid := len(line) / 2
		first := line[0:mid]
		second := line[mid:]
		ch := getCommonLetter(first, second)
		sum += getPriority(ch)
	}

	fmt.Println(sum)
}
